sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,171 @@
+        partitions_ownership_dict = dict()
+        for ownership in partitions_ownership:
+            partitions_ownership_dict[ownership["partition_id"]] = ownership
+
+        to_claim_list = []
+        for pid in self.partition_ids:
+            p_ownership = partitions_ownership_dict.get(pid)
+            if p_ownership:
+                to_claim_list.append(p_ownership)
+            else:
+                new_ownership = dict()
+                new_ownership["eventhub_name"] = self._eventhub_name
+                new_ownership["consumer_group_name"] = self._consumer_group_name
+                new_ownership["instance_id"] = self._instance_id
+                new_ownership["partition_id"] = pid
+                new_ownership["owner_level"] = 1  # will increment in preview 3
+                to_claim_list.append(new_ownership)
+        claimed_list = await self._partition_manager.claim_ownership(to_claim_list)
+        return claimed_list
+
+    async def _start_claimed_partitions(self, claimed_partitions):
+        consumers = []
+        for partition in claimed_partitions:
+            partition_id = partition["partition_id"]
+            offset = partition.get("offset")
+            offset = offset or self._initial_event_position
+            consumer = self._eventhub_client.create_consumer(self._consumer_group_name, partition_id,
+                                                            EventPosition(str(offset)))
+            consumers.append(consumer)
+
+            partition_processor = self._partition_processor_callable(
+                eventhub_name=self._eventhub_name,
+                consumer_group_name=self._consumer_group_name,
+                partition_id=partition_id,
+                checkpoint_manager=CheckpointManager(partition_id, self._eventhub_name, self._consumer_group_name,
+                                                     self._instance_id, self._partition_manager)
+            )
+            loop = asyncio.get_running_loop()
+            task = loop.create_task(
+                _receive(consumer, partition_processor, self._receive_timeout, self._cancellation_token))
+            self._tasks.append(task)
+
+        await asyncio.gather(*self._tasks)
+        await self._partition_manager.close()
+        log.info("EventProcessor %r partition manager is closed", self._instance_id)
+        log.info("EventProcessor %r partition has stopped", self._instance_id)
+
+
+async def _receive(partition_consumer, partition_processor, receive_timeout, cancellation_token):
+    async with partition_consumer:
+        while not cancellation_token.is_cancelled:
+            try:
+                events = await partition_consumer.receive(timeout=receive_timeout)
+            except EventHubError as eh_err:
+                if eh_err.error == "link:stolen":
+                    reason = CloseReason.LEASE_LOST
+                else:
+                    reason = CloseReason.EVENTHUB_EXCEPTION
```

Comment:
we can make it shorter :)
```python
reason = CloseReason.LEASE_LOST if eh_err.error == "link:stolen" else CloseReason.EVENTHUB_EXCEPTION
```

Reply:
good suggestion. Very concise pythonic statement